packages/aws-cdk-lib/aws-sns/lib/subscription.ts

```diff
@@ -152,10 +160,85 @@ export class Subscription extends Resource {
       region: props.region,
       redrivePolicy: this.buildDeadLetterConfig(this.deadLetterQueue),
       subscriptionRoleArn: props.subscriptionRoleArn,
+      deliveryPolicy: props.deliveryPolicy ? this.renderDeliveryPolicy(props.deliveryPolicy, props.protocol): undefined,
     });
 
   }
 
+  private renderDeliveryPolicy(deliveryPolicy: DeliveryPolicy, protocol: SubscriptionProtocol): any {
+    if (protocol !== SubscriptionProtocol.HTTP && protocol !== SubscriptionProtocol.HTTPS) {
+      throw new Error(`Delivery policy is only supported for HTTP and HTTPS subscriptions, got: ${protocol}`);
+    }
+    const { healthyRetryPolicy, throttlePolicy } = deliveryPolicy;
+    if (healthyRetryPolicy) {
+      const delayTargetLimitSecs = 3600;
+      const minDelayTarget = healthyRetryPolicy.minDelayTarget;
+      const maxDelayTarget = healthyRetryPolicy.maxDelayTarget;
+      if (minDelayTarget !== undefined) {
+        if (minDelayTarget.toMilliseconds() % 1000 !== 0) {
```

Comment:
I think you can directly use `toSeconds()` which will check if it's a integer second

Reply:
I thought about it, but I thought this would be better as the error `toSeconds` raises doesn't have the same context we can add here?